src/tech/v3/dataset/impl/column.clj

```diff
@@ -287,13 +343,18 @@
                     (= :object col-dtype)
                     (casting/numeric-type? (dtype/get-datatype col)))
         (throw (Exception. "Non-numeric columns do not convert to doubles.")))
-      (dtype/->double-array (dtype-proto/elemwise-reader-cast col :float64))))
+      (dtype/make-container :jvm-heap :float64 (dtype-proto/elemwise-reader-cast
+                                                col :float64))))
```

Comment:
Ah, this is the cause of the second failure.  make-container returns an array-buffer; not a pure java double-array.  Array-buffers support sub-buffers and unsigned datatypes.

Reply:
Great tests are passing now.